src/common/core/corerel.js

```diff
@@ -707,88 +830,72 @@ define([
 
         this.getCollectionNames = function (node) {
             ASSERT(self.isValidNode(node));
-
-            var target = '';
-            var names = [];
+            var names = [],
+                target = '',
+                name,
+                inverseOverlays;
 
             do {
-                var child = innerCore.getProperty(innerCore.getChild(node, CONSTANTS.OVERLAYS_PROPERTY),
-                    target);
-                if (child) {
-                    for (var name in child) {
-                        if (!self.isPointerName(name) && name !== CONSTANTS.MUTABLE_PROPERTY) {
-                            name = name.slice(0, -CONSTANTS.COLLECTION_NAME_SUFFIX.length);
-                            if (self.isPointerName(name) && names.indexOf(name) < 0) {
-                                names.push(name);
-                            }
+                inverseOverlays = self.getInverseOverlayOfNode(node);
+                if (inverseOverlays[target]) {
+                    for (name in inverseOverlays[target]) {
+                        if (names.indexOf(name) === -1) {
+                            names.push(name);
```

Comment:
The inverseOverlay should only contain names that are allowed pointer names, so there is no point in running the check.